posts/2024-07-26-crates-io-development-update.md

```diff
@@ -0,0 +1,69 @@
+- e.g. <https://static.crates.io/rss/crates/serde.xml>: The latest version updates of the `serde` crate (the past 24 hours, but at least 10 versions).
+
+This will allow you to keep track of the latest crate releases and updates in your favorite RSS reader. The original GitHub issue requested a feed for all the crates you "follow" on crates.io, but we decided that per-crate feeds would be more useful for now. If you have any feedback on this feature, please let us know!
+
+
+## API token expiry notifications
+
+Our crates.io team member [@hi-rustin](https://github.com/hi-rustin) has been very active in improving our API tokens user experience. If you create an API token with an expiry date, you will now receive a notification email three days before the token expires. This will help you to remember to renew your token before it expires and your scripts stop working.
+
+Following this change, he also implemented a way to create API tokens based on existing tokens, which will make it easier to renew tokens without having to reconfigure all the permissions. The user interface on the "API tokens" settings page now shows a "Regenerate" button, which will allow you to copy/clone existing tokens with the same permissions. Similarly, the token expiry notifications will now also contain a link that directly fills in the permissions of the expiring token, so you can easily create a new token with the same permissions.
+
+
+## Query optimizations
+
+Our latest addition to the crates.io team, [@eth3lbert](https://github.com/eth3lbert), has been working on optimizing the database queries that power crates.io. He has been working on a couple of pull requests that aim to reduce the load on the database server and make the website faster for everyone. Some of the changes he has made include:
+
+- [#7865](https://github.com/rust-lang/crates.io/pull/7865): Further speed-up reverse dependencies query
+- [#7941](https://github.com/rust-lang/crates.io/pull/7941): Improve crates endpoint performance
+- [#8734](https://github.com/rust-lang/crates.io/pull/8734): Add partial index on versions table
+- [#8737](https://github.com/rust-lang/crates.io/pull/8737): Improve the performance of reverse dependencies using the `default_versions` table
+
+In some of these cases we have seen a 10-20x performance improvement. We are very happy to have him on the team and look forward to more improvements in the future!
+
+
+## New database server
+
+We have recently migrated our database servers to a new provider with more memory and faster storage. This has improved the performance of the website and allowed us to run more complex queries without running into performance issues.
+
+It was previously taking multiple seconds to load e.g. https://crates.io/crates/syn/reverse_dependencies, but now the server usually responds in significantly less than a second. We are very happy with the new servers and hope that you will notice the improved performance as well.
```

Comment:
This is definitely good progress we've made recently and should be shared with others. However, I'm not sure about including this section here as the new server contributes to only a portion of the improvement. The main benefit seems to come from query performance improvement, which should also have responded in less than a second on the previous server, or perhaps I'm mistaken. I suggest comparing the apis that leverage `recent_crate_downloads` and `version_downloads`, as `version_downloads` should experience a significant size reduction after migrating to the new server.

Overall LGTM 👍